lib/linter-javac.coffee

```diff
@@ -15,16 +15,18 @@ class LinterJavac extends Linter
   linterName: 'javac'
 
   # A regex pattern used to extract information from the executable's output.
-  regex: 'java:(?<line>\\d+): ((?<error>error)|(?<warning>warning)): (?<message>.+)[\\n\\r]'
+  regex: 'java:(?<line>\\d+): ?((?<error>error)|(?<warning>warning))?:? (?<message>.+)[\\n\\r]'
+
+  observer: null
 
   constructor: (editor) ->
     super(editor)
 
-    atom.config.observe 'linter-javac.javaExecutablePath', =>
+    @observer = atom.config.observe 'linter-javac.javaExecutablePath', =>
       @executablePath = atom.config.get 'linter-javac.javaExecutablePath'
 
   destroy: ->
-    atom.config.unobserve 'linter-javac.javaExecutablePath'
+    @observer.dispose
```

Comment:
I fixed this in a separate commit, since this was missing parens after `dispose` causing it to not actually be called, and it needed a call to `super`.

Would you mind reverting this change and then I can merge the pattern change? Thanks.